railties/lib/rails/generators/rails/app/templates/bin/setup.tt

```diff
@@ -18,7 +18,11 @@ FileUtils.chdir APP_ROOT do
 <% if using_node? -%>
 
   # Install JavaScript dependencies
-  system("yarn install --check-files")
+  case
+  when File.exist?("yarn.lock") then system("yarn install --check-files")
+  when File.exist?("pnpm-lock.yaml") then system("pnpm install --verify --lockfile")
+  when File.exist?("package-lock.json") then system("npm install")
+  end
```

Comment:
Big +1 on supporting pnpm and npm; I suspect this current implementation won't work since usually the lockfile doesn't exist when someone is likely to run `bin/setup`?

Reply:
I wonder if this could be synced up better with `jsbundling-rails` (in particular, since it introduced support for bun, pnpm and npm, e.g.: https://github.com/rails/jsbundling-rails/pull/180).

Reply:
I think the newly generated Rails app currently will have a `yarn.lock` file. So, it typically executes the `yarn install --check-files` command. And if you don't have a yarn.lock file at all, this line won't be executed because no condition is met.